app/docs/src/how-tos/create-templates.md

```diff
@@ -0,0 +1,551 @@
+        { Key: "Environment", Value: typedInputs.environment },
+        { skipIfMissing: true }
+      );
+    }
+
+    return workingSet;
+  });
+}
+```
+
+Perform one final dry-run test.
+
+```shellscript
+si template run vpc-pattern.ts --key test --input vpc-pattern-inputs.yaml --dry-run
+```
+
+### Add External Component References
+
+For templates to work across different workspaces, you need to handle component references (like AWS Credentials and Regions) as inputs rather than hardcoded IDs. Use `SubscriptionInput` for this:
+
+```typescript
+import { SubscriptionInput, TemplateContext } from "jsr:@systeminit/si";
+import { z } from "npm:zod@4";
+
+export default async function (ctx: TemplateContext) {
+  const inputSchema = z.object({
+    environment: z.string().default("dev"),
+    vpcCidr: z.string().default("10.0.0.0/16"),
+    credential: SubscriptionInput,
+    region: SubscriptionInput,
+  });
+  type Inputs = z.infer<typeof inputSchema>;
+
+  ctx.inputs(inputSchema);
+
+  ctx.search([
+    'schema:"AWS::EC2::VPC"',
+    'schema:"AWS::EC2::Subnet"',
+    'schema:"AWS::EC2::RouteTable"',
+  ]);
+
+  ctx.namePattern([
+    { pattern: /demo-(.+)/, replacement: "<%= inputs.environment %>-$1" },
+  ]);
+
+  ctx.transform(async (workingSet, inputs) => {
+    const typedInputs = inputs as Inputs;
+
+    // Update VPC CIDR block from inputs
+    const vpc = workingSet.find((c) => c.schemaName === "AWS::EC2::VPC");
+    if (vpc) {
+      await ctx.ensureAttribute(vpc, "/domain/CidrBlock", typedInputs.vpcCidr);
+    }
+
+    // Add environment tags to all components
+    for (const component of workingSet) {
+      await ctx.ensureArrayAttribute(
+        component,
+        "/domain/Tags",
+        (e) => e.subpath === "Key" && e.value === "Environment",
+        { Key: "Environment", Value: typedInputs.environment },
+        { skipIfMissing: true }
+      );
+    }
+
+    // Set credential and region subscriptions for all components
+    for (const component of workingSet) {
+      await ctx.ensureAttribute(
+        component,
+        "/secrets/AWS Credential",
+        typedInputs.credential,
+        { skipIfMissing: true }
+      );
+
+      await ctx.ensureAttribute(
+        component,
+        "/domain/extra/Region",
+        typedInputs.region,
+        { skipIfMissing: true }
+      );
+    }
+
+    return workingSet;
+  });
+}
+```
+
+Update your input file to specify how to find these components:
+
+```yaml
+environment: prod
+vpcCidr: 10.1.0.0/16
+credential:
+  kind: "search"
+  query: 'schema:"AWS Credential"'
+  path: "/secrets/AWS Credential"
+region:
+  kind: "search"
+  query: 'schema:"Region"'
+  path: "/domain/region"
+```
+
+:::info
+
+Why does this matter? When you cache a baseline, component IDs from the original workspace won't exist in the target workspace. Using `SubscriptionInput` with search queries allows the template to find the equivalent components in any workspace.
```

Comment:
This is almost right, but focuses too much on the implementation detail (the ids) rather than the concept, which is you need to reference components that exist!

Reply:
More AI junk I missed. Will fix